agent/src/beerocks/slave/ap_manager_thread.cpp

```diff
@@ -582,8 +582,24 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
                    << beerocks::utils::convert_bandwidth_to_int(
                           (beerocks::eWiFiBandwidth)request->cs_params().bandwidth);
 
-        if (request->cs_params().channel == 0)
-            LOG(DEBUG) << "Start ACS";
+        LOG_IF(request->cs_params().channel == 0, DEBUG) << "Start ACS";
+
+        // Set transmit power
+        if (request->tx_limit_valid()) {
+            ap_wlan_hal->set_tx_power_limit(request->tx_limit());
+            if (ap_wlan_hal->get_radio_info().channel == request->cs_params().channel) {
+                LOG(DEBUG) << "Setting tx power without channel switch, send CSA notification";
+                auto notification = message_com::create_vs_message<
+                    beerocks_message::cACTION_APMANAGER_HOSTAP_CSA_NOTIFICATION>(cmdu_tx);
+                if (!notification) {
+                    LOG(ERROR) << "Failed building message!";
+                    return false;
+                }
+                ap_wlan_hal->refresh_radio_info();
```

Comment:
Since nothing has changed, refreshing radio info shouldn't be needed.

Looking at it in more detail, it's even worse. Setting the TX power will not be reflected in the result of get_tx_power(), because that is using the tx power limit from the supported channels, and that, in turn, is purely based on the regulatory limits and the hardware limit, *not* on the power limit set by NL80211_ATTR_WIPHY_TX_POWER_SETTING. So I think we need to add an actual power limit setting in bwl::RadioInfo and make sure it gets filled in by refresh_radio_info(). But that's for another PR...

Reply:
It looks like @mariomaz already included the max tx power using `NL80211_FREQUENCY_ATTR_MAX_TX_POWER` in #977 - I think we should decide how we want to proceed with @mariomaz 's PR before we merge this one.
I have UGW so I can test it correctly all the way, no need to rush and merge this half way done.

Reply:
So apparently my previous comment is wrong, we don't need to rebase on #977 in order to read the tx power since it is reflected in `TxPower` when running `hostapd_cli -iwlan0 radio_info` hence available in `m_radio_info.conducted_power`. Added a commit which fixes this.